lib/lists.nix

```
@@ -403,6 +403,49 @@ rec {
   */
   concatMap = builtins.concatMap;
 
+  /**
+    Map, removing null values.
+
+    `filterMap` is a convenience function. Use it when `null` is never a
+    valid return item; otherwise, use the more flexible `concatMap`,
+    especially in generic code.
+
+    # Type
+
+    ```
+    filterMap :: (a -> (null | b)) -> [a] -> [b]
+    ```
+
+    # Examples
+    :::{.example}
+    ## `lib.lists.filterMap` usage example
+
+    ```nix
+    let f = x:
+      if "udp" == x.proto
+      then x.port
+      else null;
+    in filterMap f [
+      {
+        dns.proto  = "udp";
+        dns.port  = 53;
+      }
+      {
+        http.proto = "tcp";
+        http.port = 80;
+      }
+    ]
+    => [ 53 ]
+    ```
+
+    :::
+  */
```

```suggestion

    # Strictness

    If all of your list is used, and it has no recursion between items, ignore this section.

    Like `filterAttrs`, this function is strict in the items, which means that unlike e.g. `map`, `filterMap` can only return after evaluating each item.
    This can be a problem when accessing list items by index or with `head`, in situations where not all items are well defined, or when a list item is defined in terms of another list item in a strict manner.

  */
```

Probably more relevant for `filterAttrs` anyway, but in case it helps someone.